server/node.go

```diff
@@ -80,24 +89,50 @@ type Node struct {
 type nodeServer Node
 
 // allocateNodeID increments the node id generator key to allocate
-// a new, unique node id.
-func allocateNodeID(db *client.DB) (proto.NodeID, error) {
-	r, err := db.Inc(keys.NodeIDGenerator, 1)
+// a new, unique node id. It will retry indefinitely on retryable
+// errors.
+func allocateNodeID(db *client.DB) proto.NodeID {
+	var id proto.NodeID
+	err := retry.WithBackoff(allocRetryOptions, func() (retry.Status, error) {
+		r, err := db.Inc(keys.NodeIDGenerator, 1)
+		if err != nil {
+			status := retry.Break
+			if _, ok := err.(util.Retryable); ok {
+				status = retry.Continue
+			}
+			return status, util.Errorf("unable to allocate node ID: %s", err)
+		}
+		id = proto.NodeID(r.ValueInt())
+		return retry.Break, nil
+	})
 	if err != nil {
-		return 0, util.Errorf("unable to allocate node ID: %s", err)
+		log.Fatal(err)
```

Comment:
Why did you move this? We already had errors being passed up the stack, so it seems better to continue to do that than to add a log.Fatal at a lower level.